src/components/Shared/Audio/index.tsx

```diff
@@ -0,0 +1,177 @@
+  barRadius: 3,
+  hideScrollbar: true,
+  responsive: true,
+  height: 40,
+  normalize: true,
+  partialRender: true
+});
+
+interface Props {
+  src: string;
+  isNew?: boolean;
+  publication?: LensterPublication;
+  txn: any;
+}
+
+export const AudioPublicationSchema = object({
+  title: string().trim().min(1, { message: 'Invalid audio title' }),
+  author: string().trim().min(1, { message: 'Invalid author name' }),
+  cover: string().trim().min(1, { message: 'Invalid cover image' })
+});
+
+const Audio: FC<Props> = ({ src, isNew = false, publication, txn }) => {
+  const [playing, setPlaying] = useState(false);
+  const [duration, setDuration] = useState('00:00');
+  const audioPublication = usePublicationStore((state) => state.audioPublication);
+  const setAudioPublication = usePublicationStore((state) => state.setAudioPublication);
+
+  const id = useId();
+  const waveformRef = useRef<HTMLDivElement>(null);
+  const waveSurfer = useRef<WaveSurfer>();
+
+  const createPlayer = useCallback(async () => {
+    const WaveSurfer = (await import('wavesurfer.js')).default;
+    if (waveformRef.current) {
+      const options = getAudioPlayerOptions(waveformRef.current);
+      waveSurfer.current = WaveSurfer.create(options);
+      if (src) {
+        waveSurfer.current?.load(src);
+      }
+      waveSurfer.current.on('ready', () => {
+        setDuration(getTimeFromSeconds(waveSurfer.current?.getDuration().toString()));
+      });
+    }
+  }, [src]);
+
+  useEffect(() => {
+    createPlayer();
+    return () => {
+      if (waveSurfer.current) {
+        waveSurfer.current.destroy();
+        setAudioPublication({ author: '', cover: '', title: '', coverMimeType: '' });
+      }
+    };
+  }, [src, createPlayer, setAudioPublication]);
+
+  const handlePlayPause = () => {
+    setPlaying(!playing);
+    waveSurfer.current?.playPause();
+  };
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setAudioPublication({ ...audioPublication, [e.target.name]: e.target.value });
+  };
+
+  return (
+    <div className="border w-full overflow-hidden border-gray-200 dark:border-gray-800 rounded-xl">
+      <div className="flex flex-1 space-x-2">
+        <CoverImage
+          isNew={isNew && !txn}
+          cover={isNew ? (txn ? txn.cover : audioPublication.cover) : getThumbnailUrl(publication)}
+          setCover={(url, mimeType) =>
+            setAudioPublication({ ...audioPublication, cover: url, coverMimeType: mimeType })
+          }
+        />
+        <div className="flex py-5 px-3 flex-col justify-between w-full">
+          <div className="flex justify-between">
+            <div className="flex items-center space-x-3 w-full">
+              <div>
+                <button
+                  type="button"
+                  onClick={handlePlayPause}
+                  className="p-2.5 bg-gray-500 hover:bg-gray-600 text-white rounded-full"
+                >
+                  {playing ? (
+                    <svg
```

Comment:
**nitpick:** we can use play pause icon from heroicons

Reply:
there is sold icons for play pause missing in heroicons package 

<img width="209" alt="image" src="https://user-images.githubusercontent.com/29498872/196682116-f008739f-1bbc-49fd-9eba-ae9e31799c30.png">

Reply:
Yup they have solid icons 🤔